src/api/entities/Identity/index.ts

```diff
@@ -691,21 +706,44 @@ export class Identity extends Entity<UniqueIdentifiers, string> {
       },
     } = this;
 
-    const assembleResult = ({
-      secondaryKeys: secondaryAccounts,
-    }: PolymeshPrimitivesIdentity): PermissionedAccount[] => {
-      return secondaryAccounts.map(({ signer: rawSigner, permissions }) => ({
-        account: signatoryToAccount(rawSigner, context),
-        permissions: meshPermissionsToPermissions(permissions, context),
-      }));
+    const assembleResult = (
+      rawSecondaryKeyKeyRecord: Option<PolymeshPrimitivesSecondaryKeyKeyRecord>[],
+      accountIds: AccountId32[]
+    ): PermissionedAccount[] => {
+      return rawSecondaryKeyKeyRecord.reduce((secondaryKeys, optKeyRecord, index) => {
+        if (optKeyRecord.isSome) {
+          const account = accountIdToAccount(accountIds[index], context);
+          const keyRecord = optKeyRecord.unwrap();
+
+          if (keyRecord.isSecondaryKey) {
+            const [, permissions] = keyRecord.asSecondaryKey;
+            secondaryKeys.push({
+              account,
+              permissions: meshPermissionsToPermissions(permissions, context),
+            });
+          }
+        }
+
+        return secondaryKeys;
+      }, [] as PermissionedAccount[]);
```

Comment:
I think you can pass this as a type parameter to `reduce` instead of having to use an assertion